extension/src/content_scripts/domEnactor.js

```diff
@@ -0,0 +1,268 @@
+/*
+ * GPII Chrome Extension for Google Chrome
+ *
+ * Copyright 2017 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this license.
+ *
+ * You may obtain a copy of the license at
+ * https://github.com/GPII/gpii-chrome-extension/blob/master/LICENSE.txt
+ */
+
+/* global fluid, chrome */
+"use strict";
+
+(function ($, fluid) {
+
+    var gpii = fluid.registerNamespace("gpii");
+
+    // The main component to handle settings that require DOM manipulations.
+    // It contains various subcomponents for handling various settings.
+    fluid.defaults("gpii.chrome.domEnactor", {
+        gradeNames: ["fluid.contextAware", "fluid.viewComponent"],
+        model: {
+            // Accepted model values:
+            // contrastTheme: string,
+            // lineSpace: number,    // the multiplier to the current line space
+            // inputsLarger: boolean,
+            // selectionTheme: string,
+            // simplifiedUiEnabled: boolean,
+            // tableOfContents: boolean
+        },
+        events: {
+            onMessage: null
```

Comment:
The various use of "onMessage" both as an event name and a namespace name I think is a little unclear since it doesn't account for in which direction the message is travelling or for what purpose. This should be given a more semantic name describing either the source of the message, its direction or purpose